s3utils/localpath.go

```diff
@@ -0,0 +1,101 @@
+package s3utils
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type localPath struct {
+	raw    string
+	client *s3.S3
+}
+
+// IsDir Checks if a localPath is a directory
+func (p localPath) IsDir() (bool, error) {
+	stat, err := os.Stat(p.raw)
+	if err != nil {
+		return false, err
+	}
+
+	return stat.IsDir(), nil
+}
+
+// Exists Checks if a localPath is a directory
+func (p localPath) Exists() (bool, error) {
+	_, err := os.Stat(p.raw)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return true, err
+}
+
+// IsS3 checks if a localPath is an S3Path (it will never be)
+func (p localPath) IsS3() bool {
+	return false
+}
+
+// IsLocal checks if a localPath is a localPath (it will always be)
+func (p localPath) IsLocal() bool {
+	return true
+}
+
+// ListPathsWithPrefix lists all paths with the localPath as a prefix
+func (p localPath) ListPathsWithPrefix() ([]Path, error) {
+	filepaths := []Path{}
+	err := filepath.Walk(
+		p.raw,
+		func(filepath string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() {
+				currentPath, err := NewPath(p.client, filepath)
+				if err != nil {
+					return err
+				}
+				filepaths = append(filepaths, currentPath)
+			}
+			return nil
+		},
+	)
+	return filepaths, err
+}
+
+// Join joins suffixes to this path
+func (p localPath) Join(suffixes ...string) Path {
+	joinArgs := append([]string{p.raw}, suffixes...)
+	p.raw = path.Join(joinArgs...)
+	return p
+}
+
+// Base gets the base name of this path
+func (p localPath) Base() string {
+	return path.Base(p.raw)
+}
+
+// ToStringWithoutBucket returns a raw string path without the s3 bucket
+func (p localPath) ToStringWithoutBucket() string {
+	return p.raw
+}
+
+// WithoutPrefix returns a string representation of this path with the
+//   prefixPath (ignoring it's s3 bucket) removed from the begining. This is
```

Comment:
nit: its

Reply:
`WithoutPrefix` seems a little ambiguous to me b/c you might expect prefix to just mean `s3://`. Maybe just `WithoutBucket` or `ObjectKey`?

Reply:
`WithoutPrefix` is definitely confusing it actually removes one path from another ignoring the scheme (`s3://`) and bucket. I will just remove the method completely, it is only used in one place so I will just put the implementation there. It is actually less code and less confusing that way imo.

Reply:
Also I will rename `ToStringWithoutBucket` to `WithoutBucket` it has a `string` return type so the name is redundant.